python/samples/06.auth.oauth.bot/src/bot.py

```diff
@@ -0,0 +1,84 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+
+Description: initialize the app and listen for `message` activitys
+"""
+
+import sys
+import traceback
+
+from botbuilder.core import MemoryStorage, TurnContext
+from teams import Application, ApplicationOptions, TeamsAdapter
+from teams.auth import AuthOptions, OAuthOptions, SignInResponse
+from teams.state import ConversationState, TempState, TurnState, UserState
+
+from config import Config
+
+config = Config()
+app = Application[TurnState[ConversationState, UserState, TempState]](
+    ApplicationOptions(
+        bot_app_id=config.APP_ID,
+        storage=MemoryStorage(),
+        adapter=TeamsAdapter(config),
+        auth=AuthOptions(
+            default="graph",
+            auto=True,
+            settings={
+                "graph": OAuthOptions(
+                    connection_name=config.OAUTH_CONNECTION_NAME,
+                    title="Sign In",
+                    text="please sign in",
+                    end_on_invalid_message=True,
+                    enable_sso=True,
+                ),
+            },
+        ),
+    )
+)
+
+
+@app.message("/signout")
+async def on_sign_out(
+    context: TurnContext, state: TurnState[ConversationState, UserState, TempState]
+):
+    await app.auth.get("graph").sign_out(context, state)
+    await context.send_activity("you are now signed out...👋")
+    return False
+
+
+@app.activity("message")
+async def on_message(
+    context: TurnContext, _state: TurnState[ConversationState, UserState, TempState]
+):
+    await context.send_activity(f"you said: {context.activity.text}")
+    return False
+
+
+@app.auth.get("graph").on_sign_in_success
```

Comment:
Getting a syntax error with Python < 3.9 - expression form not supposed for decorator